File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        i,j=0,0
        res=[]
        while i<len(nums1) and j<len(nums2):
            if nums1[i]<nums2[j]:
                res.append(nums1[i])
                i+=1
            else:
                res.append(nums2[j])
                j+=1
        if i<len(nums1):
            for k in range(i,len(nums1)):
                res.append(nums1[k])
        if j<len(nums2):
            for k in range(j,len(nums2)):
                res.append(nums2[k])
        print(res)
        n=len(res)
        
        if(n%2==1):
            return (res[n//2])
        else:
            s=res[((n-1)//2)]
            y=res[(n//2)]
            return ((s+y)/2)
                
                

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int i=0;
        int j=0;
        int c=0;
        int indexel1=-1;
        int indexele2=-1;
        int n=nums1.length;
        int m=nums2.length;
        int total=(n+m);
        int i1=total/2;
        int i2=i1-1;
        while(i<n && j<m){
            if(nums1[i]<nums2[j]){
                if(c==i1) indexel1=nums1[i];
                if(c==i2) indexele2=nums1[i];
                i++;
                c++;
            }
            else{
                if(c==i1) indexel1=nums2[j];
                if(c==i2) indexele2=nums2[j];
                j++;
                c++;
            }
        }
        while(i<n){
            
                if(c==i1) indexel1=nums1[i];
                if(c==i2) indexele2=nums1[i];
                i++;
                c++;
            
        }
        while(j<m){
            
                if(c==i1) indexel1=nums2[j];
                if(c==i2) indexele2=nums2[j];
                j++;
                c++;
            
        }
        if(total%2==1){
            return indexel1;
        }
        return (double)((indexel1+indexele2)/2.0);
    }
}